### PR TITLE
Changes to the Gemma3 backbone for Embedding Gemma model 

### PR DESCRIPTION
**Pull request by laxmareddyp**
Numerics verification:

Screenshot 2026-01-31 at 5 24 17 AM

[Comparison Colab Notebook](https://colab.research.google.com/gist/laxmareddyp/51ed894611ccf9ec51c3591fd0a1e586/test-embedgemma.ipynb)

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends (TensorFlow, JAX, and PyTorch).
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md) in making these changes.
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md) in making these changes.
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

**Comment by divyashreepathihalli** [comment]
in the screenshot - can you print the actual output values to verify the numerics matching?

File: keras_hub/src/models/gemma3/gemma3_backbone.py [hunk 1]
```diff
@@ -196,6 +214,9 @@ def __init__(
         layer_norm_epsilon=1e-6,
         use_bidirectional_attention=False,
         dropout=0,
+        is_embedding_model=False,
+        pooling_intermediate_dim=None,
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/gemma3/gemma3_backbone.py:216]: does this arg need to be exposed? does this value change with different variants?

File: keras_hub/src/models/gemma3/gemma3_backbone_test.py [hunk 2]
```diff
@@ -121,6 +121,35 @@ def test_backbone_basics(self, backbone_type):
             run_quantization_check=False,
         )
 
+    def test_embedding_mode(self):
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/gemma3/gemma3_backbone_test.py]: nit : model

File: keras_hub/src/models/gemma3/gemma3_backbone_test.py [hunk 3]
```diff
@@ -121,6 +121,35 @@ def test_backbone_basics(self, backbone_type):
...
+        pooling_intermediate_dim = 32
+        init_kwargs = self.text_init_kwargs.copy()
+        input_data = self.text_backbone_input_data.copy()
+
+        init_kwargs["is_embedding_model"] = True
+        init_kwargs["embedding_dim"] = embedding_dim
+        init_kwargs["pooling_intermediate_dim"] = pooling_intermediate_dim
+
+        model = Gemma3Backbone(**init_kwargs)
+        outputs = model(input_data)
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/gemma3/gemma3_backbone_test.py]: use standardized tests

File: keras_hub/src/models/gemma3/gemma3_mean_pooling.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import keras
+from keras import ops
+
+
```
Comment (divyashreepathihalli) [inline keras_hub/src/models/gemma3/gemma3_mean_pooling.py]: we should have one layers file - like gemma3_layers.
we could rename keras_hub/src/models/gemma3/rms_normalization.py and add this to that file

File: keras_hub/src/models/gemma3/gemma3_backbone.py (hunk 1 above)
Comment (laxmareddyp) [reply keras_hub/src/models/gemma3/gemma3_backbone.py:216]: The value remains 4096 across all variants. It seems she followed our existing KerasHub conventions—similar to how we handle layer_norm_epsilon. If you prefer not to expose this as a configurable parameter, should I hard-code the value directly at line 354 in this file and remove in other declared places?

File: keras_hub/src/models/gemma3/gemma3_backbone_test.py (hunk 2 above)
Comment (laxmareddyp) [reply keras_hub/src/models/gemma3/gemma3_backbone_test.py]: Will take care of it. Thanks

File: keras_hub/src/models/gemma3/gemma3_mean_pooling.py (hunk 4 above)
Comment (laxmareddyp) [reply keras_hub/src/models/gemma3/gemma3_mean_pooling.py]: Following your suggestion, I'm thinking to merge the `interleave embeddings`, `RMS normalization,` and `mean pooling` files into a single `gemma3_layers.py. `

Are you okay with these changes or do you have any additional suggestions?

Comment (divyashreepathihalli) [reply keras_hub/src/models/gemma3/gemma3_mean_pooling.py]: Sounds good

File: keras_hub/src/models/gemma3/gemma3_backbone.py (hunk 1 above)
Comment (divyashreepathihalli) [reply keras_hub/src/models/gemma3/gemma3_backbone.py:216]: Hmmm ....okay to leave it as is

File: tools/checkpoint_conversion/convert_gemma3_embedding_checkpoints.py [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+        output_preset = args.preset
+    else:
+        if not args.source_preset or not args.output_preset:
+            parser.error(
+                "Both --source_preset and --output_preset are required if "
+                "--preset is not provided."
+            )
+        source_preset = args.source_preset
+        output_preset = args.output_preset
+
```
Comment (sachinprasadhs) [inline tools/checkpoint_conversion/convert_gemma3_embedding_checkpoints.py]: Make these changes inside main() function and make it using FLAGS on top like 
```
flags.DEFINE_integer("pooling_intermediate_dim", 4096, "Intermediate dimension.")
flags.DEFINE_integer("embedding_dim", 768, "Final output dimension.")
```
We don't need kaggle login here to save the preset.
Let's follow conversion script pattern of other models we have.

**Review by sachinprasadhs — COMMENTED** [review]
Thanks. Looks good. Just one change
For later stage we need to think of encode feature for embedding models to improve user experience.

File: tools/checkpoint_conversion/convert_gemma3_embedding_checkpoints.py (hunk 5 above)
Comment (laxmareddyp) [reply tools/checkpoint_conversion/convert_gemma3_embedding_checkpoints.py]: Regarding the kaggle login(). These checkpoint are loading it from kaggle , it is required to login to access the source checkpoints but anyways removed and pushed it. Thanks
